CONTRIBUTING.md

```diff
@@ -0,0 +1,280 @@
+# Contribution Style Guide
```

Comment:
Since CONTRIBUTING.md already implies it's a contribution guide, there doesn't need to be a single top-level category about contributions. Likewise, subsequent categories don't need to have "Contribution" in their names, since it's already implied by the whole being a contribution guide.

Instead, I suggest the following broad structure:
- `# General information`
- - `## Behavioural Guidelines`
- - `## IDE Setup`
- `# Style Guide`
- - the rest is good

At a later time I might include "Implementation Guide" top-level category, which describes how to write utility functions themselves, the tests and the demos. But I'd rather leave it for 23.8 release - for demos in particular, I might or might not recommend using the debug view introduced in GameMaker 2023.8, depending on how well it works out.

Reply:
Headers restructured to match your suggestion.